### ISSUE TITLE
Training start from scratch after load_model()

### ISSUE THREAD

**Issue by Ciclarion** [keras-team/keras#13682]
Hello,

I'm actually using the 2.3.1 version of Keras and having a problem. I try to resume training after having save the model with a ModelCheckpoint, however the training restart from scratch. 

I know that the save function work properly, because if i try to predict using the saved model, it works well.  

After searching on internet, i found this example that's supposed to work, but it doesn't and restart the training from scratch too. 

```
import numpy as np
from numpy.testing import assert_allclose
from keras.models import Sequential, load_model
from keras.layers import LSTM, Dropout, Dense
from keras.callbacks import ModelCheckpoint

vec_size = 100
n_units = 10

x_train = np.random.rand(500, 10, vec_size)
y_train = np.random.rand(500, vec_size)

model = Sequential()
model.add(LSTM(n_units, input_shape=(None, vec_size), return_sequences=True))
model.add(Dropout(0.2))
model.add(LSTM(n_units, return_sequences=True))
model.add(Dropout(0.2))
model.add(LSTM(n_units))
model.add(Dropout(0.2))
model.add(Dense(vec_size, activation='linear'))
model.compile(loss='mean_squared_error', optimizer='adam')

# define the checkpoint
filepath = "model.h5"
checkpoint = ModelCheckpoint(filepath, monitor='loss', verbose=1, save_best_only=True, mode='min')
callbacks_list = [checkpoint]

# fit the model
model.fit(x_train, y_train, epochs=5, batch_size=50, callbacks=callbacks_list)

# load the model
new_model = load_model("model.h5")
assert_allclose(model.predict(x_train),
                new_model.predict(x_train),
                1e-5)

# fit the model
checkpoint = ModelCheckpoint(filepath, monitor='loss', verbose=1, save_best_only=True, mode='min')
callbacks_list = [checkpoint]
new_model.fit(x_train, y_train, epochs=5, batch_size=50, callbacks=callbacks_list)
```

The output is : 

> Epoch 1/5
> 2020-01-10 00:02:02.467285: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cublas64_100.dll
> 500/500 [==============================] - 2s 3ms/step - loss: 0.3268
> 
> Epoch 00001: loss improved from inf to 0.32679, saving model to model.h5
> Epoch 2/5
> 500/500 [==============================] - 0s 463us/step - loss: 0.3059
> 
> Epoch 00002: loss improved from 0.32679 to 0.30594, saving model to model.h5
> Epoch 3/5
> 500/500 [==============================] - 0s 467us/step - loss: 0.2733
> 
> Epoch 00003: loss improved from 0.30594 to 0.27330, saving model to model.h5
> Epoch 4/5
> 500/500 [==============================] - 0s 471us/step - loss: 0.2293
> 
> Epoch 00004: loss improved from 0.27330 to 0.22931, saving model to model.h5
> Epoch 5/5
> 500/500 [==============================] - 0s 490us/step - loss: 0.1872
> 
> Epoch 00005: loss improved from 0.22931 to 0.18723, saving model to model.h5
> Epoch 1/5
> 500/500 [==============================] - 1s 2ms/step - loss: 0.1545
> 
> Epoch 00001: loss improved from inf to 0.15454, saving model to model.h5
> Epoch 2/5
> 500/500 [==============================] - 0s 487us/step - loss: 0.1303
> 
> Epoch 00002: loss improved from 0.15454 to 0.13031, saving model to model.h5
> Epoch 3/5
> 500/500 [==============================] - 0s 471us/step - loss: 0.1173
> 
> Epoch 00003: loss improved from 0.13031 to 0.11731, saving model to model.h5
> Epoch 4/5
> 500/500 [==============================] - 0s 480us/step - loss: 0.1092
> 
> Epoch 00004: loss improved from 0.11731 to 0.10920, saving model to model.h5
> Epoch 5/5
> 500/500 [==============================] - 0s 485us/step - loss: 0.1063

As we can see, when the training restart after loading the model :  loss improved from inf to 0.15454, saving model to model.h5 => It start from inf, and not from the last val_loss.

However, i also discover that if i comment the two last line before fitting the new model (so the two lines that define the callback), then the training start this time from the last val_loss, as expected. 

So it's seems the problem come from the callback ? But then, if i want to close and re-open python i've no choice than redefining the call back, so... any idea ?
